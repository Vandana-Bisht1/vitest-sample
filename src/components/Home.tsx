import { UserGreeting } from "./UserGreeting";

const Home = () => {
  return (
    <div>
      <h2>This is home component</h2>
      <p data-testid="para">This is the paragraph in home page</p>
      <UserGreeting />
    </div>
  );
};
export default Home;