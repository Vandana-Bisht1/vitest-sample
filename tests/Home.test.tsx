import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import Home from "../src/components/Home"; // Ensure the path is correct
import React from "react";

describe.concurrent("Home", () => {
  it.concurrent("check if header text exists", async () => {
    render(<Home />);
    const headingElement = screen.getByText(/This is home component/i);
    expect(headingElement).toBeInTheDocument();
  });

  it.concurrent("check if para exists", async () => {
    const paraElement = screen.getByTestId("para");
    expect(paraElement).toBeInTheDocument();
    expect(paraElement).toHaveTextContent(
      /This is the paragraph in home page/i
    );
  });

  it("Home component snapshot test", () => {
    const { asFragment } = render(<Home />);
    // Create a snapshot of the component's output
    expect(asFragment()).toMatchSnapshot();
  });
});