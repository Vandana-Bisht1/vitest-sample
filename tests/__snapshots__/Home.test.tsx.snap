// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > Home component snapshot test 1`] = `
<DocumentFragment>
  <div>
    <h2>
      This is home component
    </h2>
    <p
      data-testid="para"
    >
      This is the paragraph in home page
    </p>
    <div>
      <div>
        Hello, Guest!
      </div>
      <div>
        Gender: NA
      </div>
    </div>
  </div>
</DocumentFragment>
`;
